src/main/java/org/mule/extension/http/api/request/HttpConnectivityValidator.java

```diff
@@ -0,0 +1,209 @@
+      throws ExecutionException, InterruptedException, ResponseValidatorTypedException {
+    HttpRequest request = buildTestRequest(connectionParams);
+    Result<Object, HttpResponseAttributes> result = sendRequest(client, request);
+    validateResult(request, result);
+  }
+
+  private void validateResult(HttpRequest request, Result result) {
+    getResponseValidator().validate(result, request);
+  }
+
+  private Result<Object, HttpResponseAttributes> sendRequest(HttpExtensionClient client, HttpRequest request)
+      throws InterruptedException, ExecutionException {
+    int responseTimeoutInt = (int) responseTimeoutUnit.toMillis(responseTimeout.longValue());
+    HttpResponse response =
+        client.send(request, responseTimeoutInt, followRedirects, resolveAuthentication(client)).get();
+
+    return new HttpResponseToResult()
+        .convert(new VoidHttpRequesterCookieConfig(), muleContext, response, response.getEntity(),
+                 response.getEntity()::getContent,
+                 request.getUri());
+  }
+
+  private HttpRequest buildTestRequest(RequestConnectionParams connectionParams) {
+    String uriString = getUriString(connectionParams);
+    return HttpRequest.builder()
+        .uri(uriString)
+        .method(requestMethod)
+        .headers(toMultiMap(requestBuilder.getRequestHeaders()))
+        .queryParams(toMultiMap(requestBuilder.getRequestQueryParams()))
+        .entity(requestBuilder.buildEntity())
+        .build();
+  }
+
+  private static MultiMap<String, String> toMultiMap(Iterable<? extends KeyValuePair> asList) {
+    MultiMap<String, String> asMultiMap = new MultiMap<>();
+    asList.forEach(pair -> asMultiMap.put(pair.getKey(), pair.getValue()));
+    return asMultiMap;
+  }
+
+  private String getUriString(RequestConnectionParams connectionParams) {
+    String pathWithUriParams = replaceUriParams(requestPath, requestBuilder.getRequestUriParams());
+    return format("%s://%s:%s%s", connectionParams.getProtocol().getScheme(), connectionParams.getHost(),
+                  connectionParams.getPort(), pathWithUriParams);
+  }
+
+  private ResponseValidator getResponseValidator() {
+    return responseValidator == null ? defaultStatusCodeValidator : responseValidator;
+  }
+
+  @Override
+  public void dispose() {
+    // Added just for symmetry with initialise.
+  }
+
+  @Override
+  public void initialise() throws InitialisationException {
+    if (responseValidator != null) {
+      initialiseIfNeeded(responseValidator, true, muleContext);
+    }
+  }
+
+  private static class VoidHttpRequesterCookieConfig implements HttpRequesterCookieConfig {
+
+    @Override
+    public boolean isEnableCookies() {
+      return false;
+    }
+
+    @Override
+    public CookieManager getCookieManager() {
+      return null;
+    }
+  }
+
+  private static HttpAuthentication resolveAuthentication(HttpExtensionClient client) {
+    HttpRequestAuthentication authentication = client.getDefaultAuthentication();
```

Comment:
why this? what happens if another method is used?

Reply:
Copied from here:
https://github.com/mulesoft/mule-http-connector/blob/1f64ef1c5947e8c1acba182f2bd2e579186089a2/src/main/java/org/mule/extension/http/internal/request/HttpRequester.java#L283

It's done to cast the interface HttpRequestAuthentication to HttpAuthentication, only if the object implements both. HOWEVER, it should ask if the object is instance of HttpAuthentication instead of UsernamePasswordAuthentication.
I'll do the change both here and in HttpRequester.